mlflow/pyfunc/__init__.py

```diff
@@ -964,11 +1150,11 @@ def udf_with_default_cols(*args):
                         error_code=INVALID_PARAMETER_VALUE,
                     )
             else:
-                _logger.warning(
+                raise MlflowException(
                     "Attempting to apply udf on zero columns because no column names were "
-                    "specified as arguments or inferred from the model signature."
+                    "specified as arguments or inferred from the model signature.",
+                    error_code=INVALID_PARAMETER_VALUE
                 )
```

Comment:
Note: Raise error here instead of log warning.

UDF on zero columns make no sense.
and in this PR we have to use "pandas iteration UDF" which requires at least one column input.